modules/table/src/main/java/org/apache/ignite/internal/table/distributed/disaster/DisasterRecoveryManager.java

```diff
@@ -1100,9 +1203,45 @@ private void handleTriggerKeyUpdate(WatchEvent watchEvent) {
     private void handleMessage(NetworkMessage message, ClusterNode sender, @Nullable Long correlationId) {
         if (message instanceof LocalPartitionStatesRequest) {
             handleLocalPartitionStatesRequest((LocalPartitionStatesRequest) message, sender, correlationId);
+        } else if (message instanceof LocalTableStateRequest) {
+            handleLocalTableStateRequest((LocalTableStateRequest) message, sender, correlationId);
         }
     }
 
+    private void handleLocalTableStateRequest(LocalTableStateRequest request, ClusterNode sender, @Nullable Long correlationId) {
+        assert correlationId != null : "request=" + request + ", sender=" + sender;
+
+        int catalogVersion = request.catalogVersion();
+
+        Set<ZonePartitionId> requesedPartitions = request.zonePartitionIds().stream()
+                .map(ZonePartitionIdMessage::asZonePartitionId)
+                .collect(toSet());
+
+        catalogManager.catalogReadyFuture(catalogVersion).thenRunAsync(() -> {
+            List<LocalTableStateMessage> statesList = new ArrayList<>();
+
+            raftManager.forEach((raftNodeId, raftGroupService) -> {
+                if (raftNodeId.groupId() instanceof ZonePartitionId) {
+
+                    LocalTableStateMessage message = handleSizeRequestForTable(
```

Comment:
It's messy, we request size for table as method states, but propagate zonePartitionsIds as parameter.